application-forum-ui/src/main/resources/ForumCode/ForumSheet.xml

```diff
@@ -55,14 +55,15 @@
   #elseif($xcontext.action == 'view')
     #initForum($spaceName)
     ## Depending on XE version, we consider parent-child relationship for Pre NS or space-subspace for NS.
-    #set($hierarchyCondition = "doc.parent=:forumDoc or doc.space=:forumSpace")
-    #set($topics = $services.query.xwql("from doc.object(ForumCode.TopicClass) as topic where $hierarchyCondition").bindValue('forumDoc', $doc.fullName).bindValue('forumSpace', $doc.space).execute())
+    #set($hierarchyCondition = 'doc.parent=:parent or doc.space=:space')
+    #set($topics = $services.query.xwql("from doc.object(ForumCode.TopicClass) as topic where $hierarchyCondition").bindValue('parent', $doc.fullName).bindValue('space', $doc.space).execute())
   ## Count comments
```

Comment:
Side note: I`m sure a comments count can be done in one query instead of doing N*M queries, one for each topic/answer pair.